src/item_pocket.cpp

```diff
@@ -938,38 +938,12 @@ ret_val<item_pocket::contain_code> item_pocket::can_contain( const item &it ) co
         }
     }
 
-    if( it.made_of( phase_id::LIQUID ) ) {
-        if( !data->watertight ) {
-            return ret_val<item_pocket::contain_code>::make_failure(
-                       contain_code::ERR_LIQUID, _( "can't contain liquid" ) );
-        }
-        if( size() != 0 && !item( contents.front() ).combine( it ) ) {
-            return ret_val<item_pocket::contain_code>::make_failure(
-                       contain_code::ERR_LIQUID, _( "can't mix liquid with contained item" ) );
-        }
```

Comment:
Is it safe to move this check after the restriction-based checks?  I think this will allow putting two different liquids in a container when both fit the restriction, which we probably don't want.

Reply:
It safe, because the ammo match checked. Magazine can't contain different ammo, so different liquids too.
```cpp
// the ammo must match what's inside
        if( !contents.empty() ) {
            if( it_ammo != contents.front().ammo_type() ) {
                return ret_val<item_pocket::contain_code>::make_failure(
                           contain_code::ERR_AMMO, _( "item is not the correct ammo type" ) );
            } else {
                internal_count = contents.front().count();
            }
        }
```

Reply:
That code only applies to things with an ammo_restriction; what about things with an item_restriction?  The OP says that now also overrides watertight, but I don't see code changes to that effect.

Reply:
As a side note: I don't think it's as important to make this prioritization change since #42022.

Reply:
> what about things with an item_restriction?

I have not changed anything about `item_restriction`.

>As it doing item_restriction already (but this is not documented).

I added

>(and watertight, airtight ignores too). Overrides only if holster is true.

I mean this code:
```cpp
    if( !data->item_id_restriction.empty() &&
        data->item_id_restriction.count( it.typeId() ) == 0 ) {
        return ret_val<item_pocket::contain_code>::make_failure(
                   contain_code::ERR_FLAG, _( "holster does not accept this item type" ) );
    }
    if( data->holster && !contents.empty() ) {
        item item_copy( contents.front() );
        if( item_copy.combine( it ) ) {
            return ret_val<item_pocket::contain_code>::make_success();
```
But now I see that in fact the check is skipped, only if the holster contains the same object, which obviously has already passed the checks. I will delete my note about `item_restriction` from doc.

Reply:
> I don't think it's as important to make this prioritization change since #42022.

I think this is more intuitive logic. volume\weight\length not checked when specifying ammo. There is no need to check these parameters too. This will save from future mistakes. It also requires less JSON writing.